tests/utils/test_price_utils.py

```diff
@@ -0,0 +1,577 @@
+            {
+                "target_liquidity_usd": 10000000, # Targeting 5M USD liquidity
+                "market_price": 500, # Market price of base asset
+                "apr": 0.01, # fixed rate APR you'd get from purchasing bonds; r = 0.03
+                "days_remaining": 91.25, # 3 months remaining; t = 0.25
+                "time_stretch": 36.97812836141986,
+                "init_share_price": 1.5, # original share price when pool started
+                "share_price": 2, # share price of the LP in the yield source
+                "expected_base_asset_reserves": 19186.027487682495, #
+                "expected_token_asset_reserves": 816.0074435982989, #
+                "expected_total_liquidity": 20000.0 # base > token reserves?
+            },
+            # test 6: 10M target_liquidity; 500 market price; 6% APR;
+            #   3mo remaining; 36.97812836141986 time_stretch (targets 3% APR);
+            #   1.5 init share price; 2 share price
+            {
+                "target_liquidity_usd": 10000000, # Targeting 5M USD liquidity
+                "market_price": 500, # Market price of base asset
+                "apr": 0.06, # fixed rate APR you'd get from purchasing bonds; r = 0.03
+                "days_remaining": 91.25, # 3 months remaining; t = 0.25
+                "time_stretch": 36.97812836141986,
+                "init_share_price": 1.5, # original share price when pool started
+                "share_price": 2, # share price of the LP in the yield source
+                "expected_base_asset_reserves": 5195.968749573127, #
+                "expected_token_asset_reserves": 15026.091719183272, #
+                "expected_total_liquidity": 20000.0 # base > token reserves?
+            }
+        ]
+
+        for test_case in test_cases:
+            base_asset_reserves, token_asset_reserves, total_liquidity = price_utils.calc_liquidity(
+                test_case["target_liquidity_usd"],
+                test_case["market_price"],
+                test_case["apr"],
+                test_case["days_remaining"],
+                test_case["time_stretch"],
+                test_case["init_share_price"],
+                test_case["share_price"]
+            )
+
+            np.testing.assert_almost_equal(
+                base_asset_reserves,
+                test_case["expected_base_asset_reserves"],
+                err_msg="unexpected base_asset_reserves"
+            )
+            np.testing.assert_almost_equal(
+                token_asset_reserves,
+                test_case["expected_token_asset_reserves"],
+                err_msg="unexpected token_asset_reserves"
+            )
+            np.testing.assert_almost_equal(
+                total_liquidity,
+                test_case["expected_total_liquidity"],
+                err_msg="unexpected total_liquidity"
+            )
+
+
+    # ### Spot Price and APR ###
+
+
+    def test_calc_apr_from_spot_price(self):
```

Comment:
price > 1.0?  If this breaks it, then have the expected result handle the error case.